app/Services/Apis/MarketingAPI.php

```diff
@@ -0,0 +1,71 @@
+<?php namespace services\apis;
+/**
+ * Copyright 2024 OpenStack Foundation
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+use GuzzleHttp\Client;
+use Exception;
+use GuzzleHttp\Exception\RequestException;
+use Illuminate\Support\Facades\Config;
+use Illuminate\Support\Facades\Log;
+use models\summit\Summit;
+/**
+ * Class MarketingAPI
+ * @package services\apis
+ */
+final class MarketingAPI implements IMarketingAPI
+{
+    /**
+     * @param string $api_url
+     * @param array $params
+     * @return mixed
+     * @throws Exception
+     */
+    protected function getEntity(string $api_url, array $params)
+    {
+        try {
+            $client = new Client();
+
+            foreach ($params as $param => $value) {
+                $query[$param] = $value;
+            }
+
+            $response = $client->get($api_url, ['query' => $query]);
+
+            if ($response->getStatusCode() !== 200)
+                throw new Exception('invalid status code!');
+
+            $content_type = $response->getHeaderLine('content-type');
+
+            if (empty($content_type))
+                throw new Exception('invalid content type!');
+
+            if ($content_type !== 'application/json')
+                throw new Exception('invalid content type!');
+
+            $json = $response->getBody()->getContents();
+            return json_decode($json, true);
+        }
+        catch(RequestException $ex){
+            Log::warning($ex->getMessage());
+            throw $ex;
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getConfigValues(int $summit_id, string $search_pattern, int $page = 1, int $per_page = 100): array {
+        $base_url = Config::get('marketing.base_url') ?? '';
```

Comment:
this should be a member variable populated on constructor and throw a valdiation exception if not set

Reply:
done